### PR TITLE
[Tensor Operants & Prim-Relevant] Replace multiply operants with scale

### PR DESCRIPTION
**Pull request by jiahy0825**
### PR types
Function optimization

### PR changes
Others

### Describe
Pcard-66989

In order to save memory, replace `multiply` operants with scale.

### CODE REVIEW

File: paddle/fluid/prim/api/auto_code_generated/tensor_operants_gen.py [hunk 1]
```diff
@@ -102,7 +102,7 @@ class EagerTensorOperants : public TensorOperantsBase {
 }
 
 Tensor EagerTensorOperants::multiply(const Tensor& x, const Scalar& y) {
-  return ::multiply_ad_func(x, ::full_like_ad_func(x, y));
+  return ::scale_ad_func(x, y, 0.0f, true);
```
Comment (JiabinYang) [inline paddle/fluid/prim/api/auto_code_generated/tensor_operants_gen.py:105]: how about fp64?

**Review by JiabinYang — COMMENTED** [review]
one comment

**Review by zyfncg — APPROVED** [review]

File: paddle/fluid/prim/api/auto_code_generated/tensor_operants_gen.py (hunk 1 above)
Comment (jiahy0825) [reply paddle/fluid/prim/api/auto_code_generated/tensor_operants_gen.py:105]: image

`scale_ad_func` only supports `bias` with float datatype.
`multiply` doesn't need to handle the `bias` parameter. It only needs `Tensor x` and `Scalar scale`

**Review by JiabinYang — APPROVED** [review]
LGTM

**Review by heavyrain-lzy — APPROVED** [review]

**Review by YuanRisheng — APPROVED** [review]

**Pull request merged**